includes/admin/class-amp-editor-blocks.php

```diff
@@ -123,6 +132,15 @@ public function whitelist_block_atts_in_wp_kses_allowed_html( $tags, $context )
 	 */
 	public function enqueue_block_editor_assets() {
 
+		$amp_agnostic_slug = 'amp-agnostic-blocks-compiled';
+		wp_enqueue_script(
```

Comment:
I don't think this should be in a separate script. It should rather be part of `amp-editor-blocks`, or even as part of `amp-editor-blocks-build`. The filtering out of non-AMP native blocks could be done in JS instead, as the ideal is for all AMP components to be usable in non-AMP pages this year anyway.

I don't feel strongly about this, however.

Reply:
Good point, https://github.com/ampproject/amp-wp/pull/1916/commits/55d9d5ad2ae192a0e80504af340b3f2202872bb4 and https://github.com/ampproject/amp-wp/pull/1916/commits/018714479b4345cfce0d0870d8155dcf34a8040c remove that separate script, and [unregister](https://github.com/ampproject/amp-wp/pull/1916/commits/018714479b4345cfce0d0870d8155dcf34a8040c#diff-7ac3542e9bd216dc31cffe9828cae472R926) AMP-dependent blocks using [a blacklist approach](https://wordpress.org/gutenberg/handbook/designers-developers/developers/filters/block-filters/#using-a-blacklist) that the documentation recommends.

However if there was already an AMP-dependent block on the page, and it gets unregistered, there's an error.

I need to look at this more.

For example:
1. In Native AMP, add an `amp-mathml` block to some post
2. Change to Classic mode
3. Refresh the post's block editor
4. This appears:
>TypeError: Cannot read property 'title' of undefined at editor.min.js?ver=9.0.11:55

<img width="939" alt="js-error" src="https://user-images.githubusercontent.com/4063887/54023590-f8e30380-415a-11e9-8b8e-7783155cc881.png">